src/gpio.rs

```diff
@@ -238,7 +238,10 @@ macro_rules! impl_hal_output_pin {
 
         impl embedded_hal::digital::v2::StatefulOutputPin for $pxi<$mode> {
             fn is_set_high(&self) -> Result<bool, Self::Error> {
-                Ok(unsafe { gpio_get_level($pxi::<$mode>::runtime_pin()) } != 0)
```

Comment:
Ha!! I was working on almost the same solution, except being unnecessary more complicated! :)

Anyway, following issue: at least on esp32 (and I suspect on esp32s2 and esp32s3) the pins are actually > 32, so for pins with number >= 32 you have to do the same shifting trick but against GPIO_OUT1_REG, and subtract 32 from the pin before doing the shift, as described [here](https://esp32.com/viewtopic.php?t=5793#p38567).

Reply:
You can conditionalize the code for pins > 32 by using `#[cfg(not(esp32c3))]` I guess.

Reply:
Ahh my mistake, I only tried lower-numbered GPIOs so I guess that's why it worked :)

I'll get it fixed up soon, thanks for pointing that out.